app/frontend/Shared/ContextMenu.ts

```diff
@@ -283,3 +287,32 @@ export function download(url: URLString, window: any, howSaveAsDialog: boolean,
   a.click();
   a.href = "";
 }
+// #else
+// @ts-ignore ts2300
+export async function download(url: URLString, win: any, howSaveAsDialog: boolean, filename?: string) {
+  let urlObj = new URL(url);
+  let blob: Blob;
+  if (urlObj.protocol == "http:" || urlObj.protocol == "https:") {
+    let res = await fetch(urlObj.href);
+    filename = urlObj.pathname.split("/").pop();
+    blob = await res.blob();
+  } else if (urlObj.protocol == "data:") {
+    blob = await dataURLToBlob(url);
+  } else {
+    throw new Error("Unsupported protocol: " + urlObj.protocol);
+  }
+  let ext = fileExtensionForMIMEType(blob.type);
+  filename = sanitize.filename(filename, `${filename ?? "untitled"}.${ext}`);
+  if (howSaveAsDialog) {
+    let a = document.createElement("a");
+    a.href = url;
+    a.setAttribute("target", "_blank"); // for non-same-origin images
+    a.setAttribute("download", filename);
+    a.click();
+    a.href = "";
+  } else {
+    let filePath = await appGlobal.remoteApp.path.join(await appGlobal.remoteApp.getHomeDir(), "Downloads", filename);
+    await appGlobal.remoteApp.writeFile(filePath, 0o666, await blob.arrayBuffer());
```

Comment:
```suggestion
    await appGlobal.remoteApp.writeFile(filePath, 0o644, await blob.arrayBuffer());
```

You made the files world-writable